File: frontend/eskap_app/lib/models/review.dart
import 'package:equatable/equatable.dart';

class Review extends Equatable {
  final int reviewId;
  final String userId;
  final String text;
  final double rate;
  final String date;

  final bool isOwner;

  Review(
      {this.reviewId,
      this.userId,
      this.text,
      this.rate,
      this.date,
      this.isOwner});

  @override
  String toString() {
    return 'Review : $reviewId | $userId | $text | $rate | $date | ${isOwner.toString}';
  }

  static Review fromJson(review, userId) {
    var isOwner = (userId == review['userId']) ? true : false;
    return Review(
        reviewId: review['reviewId'],
        userId: review['userId'],
        text: review['text'],
        rate: review['rate'],
        date: review['date'],
        isOwner: isOwner);
  }

  Map<String, dynamic> toJson() =>
      {"userId": userId, "text": text, "rate": rate, "date": date};

  static Review updateReviewFromPrevious(
      Review review, String userId, String date, bool isOwner) {
    return Review(
        reviewId: review.reviewId,
        userId: userId,
        text: review.text,
        rate: review.rate,
        date: date,
        isOwner: isOwner);
  }

  @override
  List<Object> get props => [reviewId, userId, text, rate, date, isOwner];
}

File: backend/escape-game-api/src/main/java/com/ustl/eskap/app/bo/eskap/Review.java
package com.ustl.eskap.app.bo.eskap;

import javax.persistence.Embeddable;

@Embeddable
public class Review {

    private long reviewId;
    private String userId;
    private String text;
    private double rate;
    private String date;

    public Review () {

    }

    public Review(long reviewId, String userId, String text, double rate, String date) {
        this.reviewId = reviewId;
        this.userId = userId;
        this.text = text;
        this.rate = rate;
        this.date = date;
    }

    public long getReviewId() {
        return reviewId;
    }

    public void setReviewId(long reviewId) {
        this.reviewId = reviewId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public double getRate() {
        return rate;
    }

    public void setRate(double rate) {
        this.rate = rate;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }
}
